### ISSUE TITLE
wr.s3.read_csv() should support single-file paths more explicitly

### ISSUE THREAD

**Issue by cshumaker-irb** [awslabs/aws-data-wrangler#1347]
**Is your idea related to a problem? Please describe.**
I ran into a weird issue trying to read CSV files generated by Athena while using the awswrangler.s3.read_csv() method. I kept getting errors about file encodings. I tested the CSV files and opened support cases and of course, they were in 'UTF-8'. However, after some debugging, I found that the awswrangler library was also finding the metadata file generated by Athena, switching into multi-file read mode, and then throwing an error upon attempting to read the binary metadata file as a unicode CSV.

**Describe the solution you'd like**
I think the s3.read_csv method should assume a single file path when using the path parameter and a new parameter, "prefix", should be added. Until then, documentation for the method should be updated to warn users of this potential mixup. Although the docs are literally correct, they are not intuitive. 

It might make sense to add a different CSV method to the Athena module such as "wr.athena.read_results(execution_id: str) -> pd.DataFrame" to better match the semantics for users who want to retrieve query results when the query execution was executed outside of awswrangler which is often the case in distributed systems. This alone would not solve the issue of a user finding the s3.read_csv method and assuming that there are no files with a prefix including file extension ".csv" other than the data file.

**Comment by kukushking** [comment]
@cshumaker-irb Can you clarify what you mean by "assume a single file path"? If you pass a complete object key it will read that object, If you pass a prefix, it will read all objects under that prefix - I think that's a fair intuitive expectation. 

You can ignore objects (like Athena metadata that's causing issues in your case) by using `path_ignore_suffix` parameter and filter suffixes using `path_suffix`. 

I do agree a method retrieving query results by an execution id makes sense, I think we can add a wrapper around existing `wr.athena._fetch_csv/parquet_result` to fetch query metadata by query id and then return the result.

**Comment by cshumaker-irb** [comment]
Hi @kukushking, thanks for looking into this. Between a quick note in the documentation and the athena.read_execution_results(), I doubt anyone will run into this problem. 

The reason I ran into it is that I wasn't thinking about the metadata files while starting to work with the read_csv() method and while trying to debug the error, was trying to understand why the CSV would cause a decoding error (it wasn't, it was the metadata file). I think a lot of users probably aren't going to think about the metadata file but I don't know how many are using read_csv on Athena results anyway.

WRT to "assume a single file path", perhaps that's the right solution. What I meant was that the path variable shouldn't be treated as either a prefix or a list of keys. Maybe the better solution would be to have a "keys" argument and a "prefix" argument, rather than one argument that captures both because it might prompt users to check if there's other files. 

I'm not sure the path_ignore_suffix and path_suffix are the right solution. They will probably do the job but not intuitv

I've solved the problem on my side by just wrapping the key in a list so I don't really need this to be implemented a certain way, I just want to spare future users a couple hours.
